src/zabapgit_repo_impl.prog.abap

```diff
@@ -295,6 +307,20 @@ CLASS lcl_repo_online IMPLEMENTATION.
 
   ENDMETHOD.  " rebuild_local_checksums.
 
+  METHOD delete_initial_online_repo.
+
+    IF me->is_offline( ) = abap_false AND me->get_sha1_local( ) IS INITIAL.
+
+      lcl_app=>repo_srv( )->delete( me ).
+
+      IF iv_commit = abap_true.
+        COMMIT WORK.
+      ENDIF.
```

Comment:
if possible, can we make the highest level responsible for commit? in the LCL_SERVICES_REPO->CLONE method there is already a COMMIT which might can be used